### ISSUE TITLE
*gesdd for empty matrix

### ISSUE THREAD

**Issue by chtobler** [Reference-LAPACK/lapack#187]
Hi,

Just an inconsistent edge case: When calling *gesdd with an m-by-0 (0-by-n) matrix and jobz='A', it should write an m-by-m identity matrix into U (an n-by-n identity matrix into VT).

Best, Christine

**Comment by langou** [comment]
I am OK with the change. And I could apply it. I suggest we wait a few days for feedback from others. 

The behavior of LAPACK when a dimension is 0 is not clear. 

We could argue both ways. Yes returning an m-by-m identity matrix makes sense. It also does make sense to have a quick return where nothing gets done when a dimension is 0. 

Mathematically the most pleasant solution is ``returning an m-by-m identity matrix``. Practically, it might be better to do nothing and have a quick return. 

Let us wait a few days for feedback from others.

**Comment by mgates3** [comment]
This would of course affect all the SVD routines (gesvd, gesdd, gesvj, gejsv, gesvdx). I'm fine keeping the current behavior (quick return, do nothing with m = n = 0).

**Comment by langou** [comment]
Good point Mark. I start to think that this should be handled at a higher level than LAPACK. We can also have the same kind of sought behavior for some routines like ORGQR for example. I start to change my opinion that should be handled at a higher level than LAPACK. 

Feedback from other welcome.

**Comment by chtobler** [comment]
It's certainly possible to work around the empty case when calling into this function, and not really a problem for me. The danger I see is that someone could pass in uninitialized memory, and blindly assume that the matrix pointed to by U is orthogonal and can be used as such. This could cause some hard-to-find bugs.

 If you decide to keep the behavior as is, it would be helfpul to document in the help text for U and VT that they are only written on if min(m, n) is greater than 0.
